elephant/spike_train_correlation.py

```diff
@@ -146,3 +148,197 @@ def corrcoef(binned_sts, binary=False):
+        have positive IDs, e.g.,:
+        np.array([-3, -2, -1, 0, 1, 2, 3])
+
+    Example
+    -------
+        Plot the cross-correlation histogram between two Poisson spike trains
+
+        >>> binned_st1 = elephant.conversion.BinnedSpikeTrain(
+                elephant.spike_train_generation.homogeneous_poisson_process(
+                    10. * pq.Hz, t_start=0 * pq.ms, t_stop=2000 * pq.s),
+                binsize=1. * pq.ms)
+        >>> binned_st2 = elephant.conversion.BinnedSpikeTrain(
+                elephant.spike_train_generation.homogeneous_poisson_process(
+                    10. * pq.Hz, t_start=0 * pq.ms, t_stop=2000 * pq.s),
+                binsize=1. * pq.ms)
+
+        >>> cc_hist = cross_correlation_histogram(
+                binned_st1, binned_st2, window=20,
+                normalize=True, border_correction=False,
+                binary=False, kernel=None)
+
+        >>> plt.bar(
+                left=cc_hist[0].times.magnitude,
+                height=cc_hist[0][:, 0].magnitude,
+                width=cc_hist[0].sampling_period.magnitude)
+        >>> plt.xlabel('time (' + str(cc_hist[0].times.units) + ')')
+        >>> plt.ylabel('normalized cross-correlation histogram')
+        >>> plt.axis('tight')
+        >>> plt.show()
+
+    Alias
+    -----
+    cch
+    """
+    if st1.binsize != st2.binsize:
+        raise ValueError(
+            "Input spike trains must be binned with the same bin size")
+
+    # Retrieve unclipped matrix
+    st1_spmat = st1.to_sparse_array()
+    st2_spmat = st2.to_sparse_array()
+
+    # For each row, extract the nonzero column indices and the corresponding
+    # data in the matrix (for performance reasons)
+    st1_bin_idx_unique = st1_spmat.nonzero()[1]
+    st2_bin_idx_unique = st2_spmat.nonzero()[1]
+    if binary:
+        st1_bin_counts_unique = np.array(st1_spmat.data > 0, dtype=int)
+        st2_bin_counts_unique = np.array(st2_spmat.data > 0, dtype=int)
+    else:
+        st1_bin_counts_unique = st1_spmat.data
+        st2_bin_counts_unique = st2_spmat.data
+
+    # Define the half-length of the full crosscorrelogram
+    #
+    # TODO: What is correct here? Why +, not max? How can we have an entry
+    # beyond the maximum length of the array?
+    hist_half_length = np.max([st1.num_bins, st2.num_bins]) - 1
+    hist_length = 2 * hist_half_length + 1
+    # hist_length = st1.num_bins + st2.num_bins - 1
+    # hist_half_length = hist_length // 2
+    if window is None:
+        hist_bins = hist_half_length
+    else:
+        hist_bins = min(window, hist_half_length)
+
+    # Initialize the counts to an array of zeroes, and the bin IDs to integers
+    # spanning the time axis
+    counts = np.zeros(2 * hist_bins + 1)
+    bin_ids = np.arange(-hist_bins, hist_bins + 1)
+    # Compute the CCH at lags in -hist_bins,...,hist_bins only
+    for r, i in enumerate(st1_bin_idx_unique):
+        timediff = st2_bin_idx_unique - i
+        timediff_in_range = np.all(
```

Comment:
This is rather inefficient, because for each spike you create a new boolean array `timediff_in_range`. Since the spikes are sorted, you could use `np.searchsorted` instead to find the indices of the first and last spike in the window.